### PR TITLE
feat(react-query): update to major v5

### PR DESCRIPTION
**Pull request by DakEnviy**

### CODE REVIEW

File: src/core/utils/composeKey.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // TODO(DakEnviy): Do not use react-query in core
```
Comment (personaljs) [inline src/core/utils/composeKey.ts:1]: Do you need todo?

File: src/react-query/impl/infinite/hooks.ts [hunk 2]
```diff
@@ -11,10 +14,10 @@ export const useInfiniteQueryData = <TDataSource extends AnyInfiniteQueryDataSou
     context: DataSourceContext<TDataSource>,
     dataSource: TDataSource,
     params: DataSourceParams<TDataSource>,
-    options?: DataSourceOptions<TDataSource>,
-) => {
+    options?: Partial<DataSourceOptions<TDataSource>>,
```
Comment (personaljs) [inline src/react-query/impl/infinite/hooks.ts:17]: Can you move partial type to DataSourceOptions?

File: src/react-query/utils/normalizeStatus.ts [hunk 3]
```diff
@@ -6,7 +6,11 @@ export const normalizeStatus = (
     status: QueryStatus,
     fetchStatus: FetchStatus,
 ): DataLoaderStatus => {
-    if (status === 'loading' && fetchStatus === 'idle') {
+    if (status === 'pending') {
+        if (fetchStatus === 'fetching') {
+            return 'loading';
```
Comment (personaljs) [inline src/react-query/utils/normalizeStatus.ts:11]: Why didn't you use `pending`?

**Review by personaljs — APPROVED** [review]

File: src/core/utils/composeKey.ts (hunk 1 above)
Comment (DakEnviy) [reply src/core/utils/composeKey.ts:1]: Yeah, just a reminder to find another function for hashing in the future

File: src/react-query/impl/infinite/hooks.ts (hunk 2 above)
Comment (DakEnviy) [reply src/react-query/impl/infinite/hooks.ts:17]: I believe that having origin options can be useful

File: src/react-query/utils/normalizeStatus.ts (hunk 3 above)
Comment (DakEnviy) [reply src/react-query/utils/normalizeStatus.ts:11]: `loading` should tell to draw a loader, it's like `isLoading` boolean flag is `useQuery` result. `pending` has a little bit different meaning

**Pull request merged**